src/configure/configure.ts

```diff
@@ -48,26 +64,38 @@ class PipelineConfigurer {
     private appServiceClient: AppServiceClient;
     private workspacePath: string;
     private uniqueResourceNameSuffix: string;
+    private telemetryHelper: TelemetryHelper;
 
-    public constructor() {
+    public constructor(telemetryHelper: TelemetryHelper) {
         this.inputs = new WizardInputs();
         this.inputs.azureSession = extensionVariables.azureAccountExtensionApi.sessions[0];
         this.azureDevOpsClient = new AzureDevOpsClient(this.inputs.azureSession.credentials);
         this.azureDevOpsHelper = new AzureDevOpsHelper(this.azureDevOpsClient);
         this.uniqueResourceNameSuffix = uuid().substr(0, 5);
+        this.telemetryHelper = telemetryHelper;
     }
 
     public async configure(node: any) {
+        this.telemetryHelper.setCurrentStep('GetAllRequiredInputs');
         await this.getAllRequiredInputs(node);
+
+        this.telemetryHelper.setCurrentStep('CreatePreRequisites');
```

Comment:
is there a way user can cancel createPreRequisites?

Reply:
Yes, he might not enter the GitHub PAT token